forum/qa-plugin/q2apro-on-site-notifications/q2apro-history-check.php

```diff
@@ -23,182 +24,150 @@
-						
-						// don't inform user that comments, and don't inform user that comments on his own question/answer
-						if($userid_CommThr != $uid && $userid_CommThr != $pid) {
-							$ohandle = qa_post_userid_to_handle($userid_CommThr);
-
-							$paramstring = '';
-							foreach ($params as $key => $value) {
-								$paramstring .= (strlen($paramstring) ? "\t" : '').$key.'='.$this->value_to_text($value);
-							}
-							
-							qa_db_query_sub(
-								'INSERT INTO ^eventlog (datetime, ipaddress, userid, handle, cookieid, event, params) '.
-								'VALUES (NOW(), $, $, $, #, $, $)',
-								qa_remote_ip_address(), $userid_CommThr, $ohandle, $cookieid, $oevent, $paramstring
-							);
-						}
-					}
-				} // end in_c_comment
-				
-			} // end in_array
-		} // end process_event
-
-
-		// worker functions
-		function value_to_text($value) {
-			if (is_array($value))
-				$text='array('.count($value).')';
-			elseif (strlen($value)>40)
-				$text=substr($value, 0, 38).'...';
-			else
-				$text=$value;
-				
-			return strtr($text, "\t\n\r", '   ');
-		}
-		
-	} // end class
-
-	
-/*
-	Omit PHP closing tag to help avoid accidental output
-*/
\ No newline at end of file
+/* The following code originates from q2a plugin "History" by NoahY and has been modified by q2apro.com
+ * It is licensed under GPLv3 http://www.gnu.org/licenses/gpl.html
+ * Link to plugin file: https://github.com/NoahY/q2a-history/blob/master/qa-history-check.php
+ */
+
+class q2apro_history_check
+{
+    function process_event($event, $userid, $handle, $cookieid, $params)
+    {
+        if (!qa_opt('event_logger_to_database')) {
+            return;
+        }
+
+        require_once QA_INCLUDE_DIR . 'qa-app-posts.php';
+
+        switch ($event) {
+            case 'a_select':
+            case 'q_vote_up':
+            case 'a_vote_up':
+            case 'q_vote_down':
+            case 'a_vote_down':
+                $this->handle_standard_event($event, $userid, $cookieid, $params);
+                break;
+            case 'a_post':
+                $this->handle_answer_comment_create($event, $userid, $cookieid, $params);
+                break;
+            case 'c_post':
+                $this->handle_answer_comment_create($event, $userid, $cookieid, $params);
+                $this->handle_comments_thread($cookieid, $params);
+                break;
+            case 'a_to_c':
+                qa_db_query_sub(
+                    'DELETE FROM ^eventlog WHERE event="in_a_question" AND params LIKE #',
+                    "postid={$params['postid']}	%"
+                );
+                $this->handle_answer_comment_create('c_post', $userid, $cookieid, $params);
+                $this->handle_comments_thread($cookieid, $params);
+                break;
+        }
+    }
+
+    private function handle_standard_event($event, $loggedUserId, $cookieId, $params)
+    {
+        if (strpos($event, 'u_') === 0) {
+            $eventUserId = $params['userid'];

```

Comment:
nie sprawdzasz czy userid istnieje

Reply:
Nie wydaje mi się, aby w praktyce mogło w tym miejscu go nie być, ale w teorii pewnie jest to możliwe, tak samo jak to, że niżej nie będzie np. postid. Ale ok, dodałem isset do warunku dla pewności.